### PR TITLE
Update debug

### PR DESCRIPTION
**Pull request by soltysh**
/assign @juanvallejo

### CODE REVIEW

**Comment by soltysh** [comment]
Fixed unit failures.

**Comment by soltysh** [comment]
Adding approved label manually, since this is bringing in some upstream patches that the bot does not pick me as approver.

File: pkg/oc/cli/cmd/debug.go [hunk 1]
```diff
@@ -222,75 +248,32 @@ func (o *DebugOptions) Complete(cmd *cobra.Command, f kcmdutil.Factory, args []s
...
-		cmdParent := cmd.Parent()
-		if cmdParent != nil {
-			fullCmdName = cmdParent.CommandPath()
-		}
-
-		if len(fullCmdName) > 0 && kcmdutil.IsSiblingCommandExists(cmd, "describe") {
-			fmt.Fprintf(o.Attach.ErrOut, "Defaulting container name to %s.\n", pod.Spec.Containers[0].Name)
-			fmt.Fprintf(o.Attach.ErrOut, "Use '%s describe pod/%s -n %s' to see all of the containers in this pod.\n", fullCmdName, pod.Name, pod.Namespace)
-			fmt.Fprintf(o.Attach.ErrOut, "\n")
+	if o.PrintFlags.OutputFormat != nil && len(*o.PrintFlags.OutputFormat) > 0 {
```
Comment (juanvallejo) [inline pkg/oc/cli/cmd/debug.go]: What if we are printing --template w/o specifying --output ?

File: pkg/oc/cli/cmd/debug.go [hunk 2]
```diff
@@ -317,27 +306,80 @@ func (o *DebugOptions) Complete(cmd *cobra.Command, f kcmdutil.Factory, args []s
...
+	if len(o.FilenameOptions.Filenames) > 0 {
+		b.FilenameParam(o.ExplicitNamespace, &o.FilenameOptions)
+	}
+	one := false
+	infos, err := b.Do().IntoSingleItemImplied(&one).Infos()
+	if err != nil {
+		return err
+	}
+	if !one {
+		return fmt.Errorf("you must identify a resource with a pod template to debug")
```
Comment (juanvallejo) [inline pkg/oc/cli/cmd/debug.go]: s/identify/specify/

File: pkg/oc/cli/cmd/debug.go [hunk 3]
```diff
@@ -317,27 +306,80 @@ func (o *DebugOptions) Complete(cmd *cobra.Command, f kcmdutil.Factory, args []s
...
+	}
+
 	names := containerNames(o.Attach.Pod)
 	if len(names) == 0 {
 		return fmt.Errorf("the provided pod must have at least one container")
 	}
-	if (o.AsRoot || o.AsNonRoot) && o.AsUser > 0 {
-		return fmt.Errorf("you may not specify --as-root and --as-user=%d at the same time", o.AsUser)
-	}
 	if len(o.Attach.ContainerName) == 0 {
```
Comment (juanvallejo) [inline pkg/oc/cli/cmd/debug.go]: could these be moved to Validate as well?

**Comment by juanvallejo** [comment]
@soltysh One comment in debug regarding --template printing. Otherwise lgtm

File: pkg/oc/cli/cmd/debug.go (hunk 2 above)
Comment (soltysh) [reply pkg/oc/cli/cmd/debug.go]: This can be fixed in followup, I didn't change the original strings.

File: pkg/oc/cli/cmd/debug.go (hunk 3 above)
Comment (soltysh) [reply pkg/oc/cli/cmd/debug.go]: I didn't want to deal with the original flow so I left is as is, for now.

**Comment by deads2k** [comment]
@juanvallejo this no longer builds.  Please pick it up and rebase it.

**Comment by juanvallejo** [comment]
@deads2k 
> this no longer builds. Please pick it up and rebase it.

On it

**Comment by deads2k** [comment]
replaced by https://github.com/openshift/origin/pull/20257

**Pull request closed without merging**